### PR TITLE
Fix issue with Properties property of PSResource

### PR DESCRIPTION
**Pull request by cormacpayne**
## Description

The `Properties` property of `PSResource` was changed from a `PSObject` to `JToken` in version 6.0.0 of `AzureRM.Resources`. This change made it difficult for users to access the key-value pairs in `Properties` and broke existing scripts trying to update generic resources (such as [this script](https://github.com/Azure/azure-powershell/blob/master/src/ResourceManager/Compute/Commands.Compute/Extension/AzureDiskEncryption/Scripts/AzureDiskEncryptionPreRequisiteSetup.ps1#L160)).

This pull request changes `Properties` to a `PSObject` by converting the `JToken` appropriately.

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by MiYanni** [comment]
On-Demand: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/839/

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/845/

**Comment by cormacpayne** [comment]
Need test coverage -- will ship this out of band

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/SdkModels/Resources/PSResource.cs [hunk 1]
```diff
@@ -62,7 +65,7 @@ public PSResource(GenericResource resource)
             this.ManagedBy = resource.ManagedBy;
             this.Name = resource.Name;
             this.Plan = resource.Plan;
-            this.Properties = resource.Properties;
+            this.Properties = ((JToken)resource.Properties).ToPsObject();
```
Comment (twitchax) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/SdkModels/Resources/PSResource.cs]: Nice.

Also, argh PowerShell, haha.

Comment (twitchax) [reply src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/SdkModels/Resources/PSResource.cs]: Also, is it ever possible for `resource.Properties` to not be convertible to `JToken`?

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/872/

**Pull request merged**